### PR TITLE
Enable the caller to select the newest, oldest-supported GHC, or both

### PR DESCRIPTION
**Pull request by Kleidukos**
With the following stanza: `tested-with: GHC ==9.10.1 || ==9.6.6 || ==8.10.7`

```
❯ cabal run -- get-tested --ubuntu-version=latest *.cabal
matrix={"include":[{"ghc":"9.10.1","os":"ubuntu-latest"},{"ghc":"9.6.6","os":"ubuntu-latest"},{"ghc":"8.10.7","os":"ubuntu-latest"}]}

❯ cabal run -- get-tested --newest --ubuntu-version=latest *.cabal
matrix={"include":[{"ghc":"9.10.1","os":"ubuntu-latest"}]}

❯ cabal run -- get-tested --oldest --ubuntu-version=latest *.cabal
matrix={"include":[{"ghc":"8.10.7","os":"ubuntu-latest"}]}

❯ cabal run -- get-tested --oldest --newest --ubuntu-version=latest *.cabal
matrix={"include":[{"ghc":"8.10.7","os":"ubuntu-latest"},{"ghc":"9.10.1","os":"ubuntu-latest"}]}
```

cc @mmhat 

relates to #52

### CODE REVIEW

File: src/GetTested/Extract.hs [hunk 1]
```diff
@@ -84,3 +90,30 @@ expandUnionVersionRange :: VersionRange -> Vector VersionRange
...
+  -- ^ Enable the newest GHC version
+  -> Bool
+  -- ^ Enable the oldest GHC version
+  -> Vector Version
+  -- ^ Input
+  -> Vector Version
+pickCompilers enableNewest enableOldest versions =
+  let getNewest =
+        if enableNewest
+          then Vector.singleton $ getDown $ Vector.maximum (fmap Down versions)
```
Comment (mmhat) [inline src/GetTested/Extract.hs:105]: Not sure what I miss here, but why do we need to wrap the versions in `Down`?

File: src/GetTested/Extract.hs [hunk 2]
```diff
@@ -84,3 +90,30 @@ expandUnionVersionRange :: VersionRange -> Vector VersionRange
...
+          else Vector.empty
+      getOldest =
+        if enableOldest
+          then Vector.singleton $ getDown $ Vector.minimum (fmap Down versions)
```
Comment (mmhat) [inline src/GetTested/Extract.hs:109]: Same here.

File: src/GetTested/Extract.hs (hunk 1 above)
Comment (mmhat) [reply src/GetTested/Extract.hs:105]: Also: `Data.Vector.maximum` is partial, so we should probably check that the vector is not empty before using it.

File: get-tested.cabal [hunk 3]
```diff
@@ -16,52 +16,96 @@ category:        Development
...
+
+common rts-options
+  ghc-options: -rtsopts -threaded "-with-rtsopts=-N -T"
+
+library
+  import:          extensions
+  import:          ghc-options
+  hs-source-dirs:  src
+
+  -- cabal-fmt: expand src/
```
[the remark this answers is not in the document]
Comment (Kleidukos) [reply get-tested.cabal:54]: I haven't been disappointed by `cabal-fmt` so far, but I'll remember that I have the option to move, thank you.

File: app/Main.hs [hunk 4]
```diff
@@ -51,38 +48,52 @@ parseOptions =
...
 
 withInfo :: Parser a -> String -> ParserInfo a
 withInfo opts desc = info (helper <*> opts) $ progDesc desc
 
-processFlag
+processOSFlag
   :: RunnerOS
   -- ^ OS flag we're processing
   -> Bool
   -- ^ explicit version
```
[the remark this answers is not in the document]
Comment (Kleidukos) [reply app/Main.hs:79]: yep

**Comment by Kleidukos** [comment]
@mmhat I have made the options mutually incompatible. Anything else you'd like to ask for?

**Comment by mmhat** [comment]
> @mmhat I have made the options mutually incompatible. Anything else you'd like to ask for?

No, LGTM :slightly_smiling_face:

**Pull request merged**